Pocket Casts Watch App Extension/EpisodeRowViewModel.swift

```diff
@@ -11,56 +11,69 @@ class EpisodeRowViewModel: EpisodeViewModel, Identifiable {
+
+    override func hydrate() {
+        if alreadyHydrated {
+            return
+        }
+        super.hydrate()
 
+        updateProperties(episode: episode, downloadProgress: downloadProgress)
         Publishers.CombineLatest($episode, $downloadProgress)
             .receive(on: RunLoop.main)
+            .dropFirst()
             .sink(receiveValue: { [unowned self] episode, downloadProgress in
-                var informationLabel = [String]()
-                var accessibilityLabel = [episode.title ?? ""]
+                updateProperties(episode: episode, downloadProgress: downloadProgress)
+            })
+            .store(in: &cancellables)
+    }
 
-                if let publishedDate = episode.publishedDate {
-                    let info = DateFormatHelper.sharedHelper.shortLocalizedFormat(publishedDate)
-                    informationLabel.append(info)
-                    accessibilityLabel.append(info)
-                }
+    private func updateProperties(episode: BaseEpisode, downloadProgress: DownloadProgress?) {
+        var informationLabel = [String]()
+        var accessibilityLabel = [episode.title ?? ""]
 
-                isDownloading = episode.downloading() || downloadProgress != nil
+        if let publishedDate = episode.publishedDate {
+            let info = DateFormatHelper.sharedHelper.shortLocalizedFormat(publishedDate)
+            informationLabel.append(info)
+            accessibilityLabel.append(info)
+        }
 
-                let info: String
-                let statusText: String?
-                switch episode.episodeStatus {
-                case DownloadStatus.downloaded.rawValue:
-                    self.downloadStatusIconName = "episodedownloaded"
-                    info = episode.displayableTimeLeft()
-                    statusText = L10n.statusDownloaded
-                case DownloadStatus.downloading.rawValue:
-                    self.downloadStatusIconName = nil
-                    info = episode.displayableInfo(includeSize: false)
-                    statusText = L10n.statusDownloading
-                case DownloadStatus.downloadFailed.rawValue:
-                    self.downloadStatusIconName = "downloadfailed"
-                    informationLabel = []
-                    accessibilityLabel = [L10n.downloadFailed]
-                    info = episode.displayableInfo(includeSize: false)
-                    statusText = nil
-                default:
-                    self.downloadStatusIconName = nil
-                    info = episode.displayableInfo(includeSize: false)
-                    statusText = L10n.statusNotDownloaded
-                }
+        isDownloading = episode.downloading() || downloadProgress != nil
 
-                informationLabel.append(info)
-                accessibilityLabel.append(info)
+        let info: String
+        let statusText: String?
+        switch episode.episodeStatus {
+        case DownloadStatus.downloaded.rawValue:
```

Comment:
I know this is just old code in a new place but we can improve readability:

```suggestion
        case DownloadStatus(rawValue: episode.episodeStatus):
```

This way you can change every `case`. So instead of:

```switch
case DownloadStatus.downloading.rawValue:
```

Just:

```swift
case .downloading:
```

Reply:
Oh, I didn't even *look* at that block of code, I didn't want to mess around with more than strictly needed :)

But that's a good change, done in https://github.com/Automattic/pocket-casts-ios/pull/950/commits/70f4cb57e6f6098bfb2a2b80bb294b4884367edf

I've also rebased with `trunk` and added a changelog entry. I'm still not sure what is changelog-worthy, but this may be noticeable for power users :)

I'll open a follow-up PR with the rest of the changes once this is merged.